caikit_nlp/data_model/classification.py

```diff
@@ -0,0 +1,32 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Data structures for text generation representations
+"""
+# Standard
+from typing import List
+
+# First Party
+from caikit.core import DataObjectBase
+
+# First party
+import alog
+import caikit
+
+log = alog.use_channel("DATAM")
+
+
+@caikit.core.dataobject(package="caikit_data_model.caikit_nlp")
+class ClassificationTrainRecord(DataObjectBase):
+    text: str
+    labels: List[str]
```

Comment:
should we expand this to `List[Union[str, int]]` ?  🤔

Reply:
sure but just mentioining internal conversion that will take place -

[label1,label2, label3] to "label1, label2, label3" because generation train record takes only string
then with int [1,2,3] it will be "1,2,3"